airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SourceSnowflakeOAuthFlow.java

```diff
@@ -141,4 +142,9 @@ private String extractUrl(JsonNode inputOAuthConfiguration) {
     return url == null ? "snowflakecomputing.com" : url.asText();
   }
 
+  private String extractRole(JsonNode inputOAuthConfiguration) {
+    var role = inputOAuthConfiguration.get("role");
+    return role == null ? "" : role.asText();
```

Comment:
should this fail if not set or is an empty role a sensible default?

Reply:
You are right, i also found this case during testing backward compatibility, fixed it with checking for not empty role field.